### PR TITLE
Developer: Guide Using AWS Services

### PR DESCRIPTION
**Pull request by skmcgrail**
This PR covers content for `content/en/making-requests.md`.

### CODE REVIEW

File: content/en/docs/configuring-sdk/_index.md [hunk 1]
```diff
@@ -67,19 +67,250 @@ export AWS_REGION=us-west-2
...
+
+```go
+cfg, err := config.LoadDefaultConfig(config.WithCredentialsProvider(customProvider))
+```
+
+If you explicitly provide credentials, as in this example, the SDK uses only those credentials.
+
+{{% pageinfo color="info" %}}
+All credential providers passed to or returned by `LoadDefaultConfig` will be wrapped in a
+[CredentialsCache]({{< apiref "aws#CredentialsCache" >}}) automatically. This allows for caching and concurrency safe
```
Comment (Doug-AWS) [inline content/en/docs/configuring-sdk/_index.md:278]: "allows for" -> "enables"

File: content/en/docs/configuring-sdk/_index.md [hunk 2]
```diff
@@ -67,19 +67,250 @@ export AWS_REGION=us-west-2
...
+```go
+cfg, err := config.LoadDefaultConfig(config.WithCredentialsProvider(customProvider))
+```
+
+If you explicitly provide credentials, as in this example, the SDK uses only those credentials.
+
+{{% pageinfo color="info" %}}
+All credential providers passed to or returned by `LoadDefaultConfig` will be wrapped in a
+[CredentialsCache]({{< apiref "aws#CredentialsCache" >}}) automatically. This allows for caching and concurrency safe 
+credential access. If you explicitly configure a provider on `aws.Config` directly you must wrap the provider
```
Comment (Doug-AWS) [inline content/en/docs/configuring-sdk/_index.md:279]: "wrap" -> "explicitly wrap" and remove the later "explicitly"

File: content/en/docs/configuring-sdk/custom-http.md [hunk 3]
```diff
@@ -28,22 +70,67 @@ alive.
...
 See <https://golang.org/pkg/net/#Dialer.KeepAlive>
 
-We'll call this ``ConnKeepAlive`` as **time.Duration**.
+Set `KeepAlive` as **time.Duration**.
 
-## Dialer.Timeout
+#### Dialer.Timeout
 
-This setting represents the maximum amount of time a dial to wait for a connection to be created.
+This setting represents the maximum amount of time a dial will wait for a connection to be created.
```
Comment (Doug-AWS) [inline content/en/docs/configuring-sdk/custom-http.md:77]: "will wait" -> "waits"

File: content/en/docs/configuring-sdk/custom-http.md [hunk 4]
```diff
@@ -28,22 +70,67 @@ alive.
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	tr.Proxy = http.ProxyURL(proxyURL)
+})
+```
+
+### Other Settings
+
+Below are a few other `Transport` settings that can be modified to tune the HTTP client. Any additional settings not
```
Comment (Doug-AWS) [inline content/en/docs/configuring-sdk/custom-http.md:114]: "Below are" -> "This section describes"

File: content/en/docs/configuring-sdk/custom-http.md [hunk 5]
```diff
@@ -28,22 +70,67 @@ alive.
...
+
+httpClient := awshttp.NewBuildableClient().WithTransportOptions(func(tr *http.Transport) {
+	tr.ExpectContinueTimeout = 0
+	tr.MaxIdleConns = 10
+})
+```
+
+#### Transport.ExpectContinueTimeout
+
+Ff the request has an "Expect: 100-continue" header, this setting represents the maximum amount of time to wait for a
```
Comment (Doug-AWS) [inline content/en/docs/configuring-sdk/custom-http.md:132]: "Ff" -> "If"

File: content/en/docs/configuring-sdk/endpoints.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+The endpoint resolver is given the service and region as a string, allowing for the resolver to dynamically drive its
+behavior. Each service client package has an exported `ServiceID` constant which can be used to determine which service
+client is invoking your endpoint resolver.
+
+An endpoint resolver can use the [EndpointNotFoundError]({{< apiref "aws#EndpointNotFoundError" >}}) sentinel error
+value to trigger fallback resolution to the service clients default resolution logic. This allows you to selectively
+override one or more endpoints seamlessly without having to handle fallback logic.
+
+If your endpoint resolver implementation returns an error other than `EndpointNotFoundError`, endpoint resolution will
+stop and the service operation will return an error to your application.
```
Comment (Doug-AWS) [inline content/en/docs/configuring-sdk/endpoints.md:32]: "will return" -> "returns"

File: content/en/docs/configuring-sdk/logging.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+The {{% alias sdk-go %}} has logging facilities available that allow your application to enable debugging information
+for debugging and diagnosing request issues or failures. The [Logger]({{< apiref smithy="logging#Logger" >}}) interface
+and [ClientLogMode]({{< apiref "aws#ClientLogMode" >}}) are the main components available to you for determining how and
+what should be logged by clients.
+
+## Logger
+
+When constructing an [Config]({{< apiref "aws#Config" >}}) using
+[LoadDefaultConfig]({{< apiref "config#LoadDefaultConfig" >}}) a default `Logger` will be configured to send log
```
Comment (Doug-AWS) [inline content/en/docs/configuring-sdk/logging.md:16]: "will be" -> "is"

File: content/en/docs/configuring-sdk/logging.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+```go
+cfg, err := config.LoadDefaultConfig(config.WithLogger(applicationLogger))
+```
+
+Now clients configured using the constructed `aws.Config` will send log messages to `applicationLogger`.
+
+### Context-Aware Loggers
+
+A Logger implementation may implement the optional [ContextLogger]({{< apiref smithy="logging#ContextLogger" >}})
+interface. Loggers that implement this interface will have its `WithContext` method invoked with the current context.
```
Comment (Doug-AWS) [inline content/en/docs/configuring-sdk/logging.md:32]: "its WithContext method" -> "their WithContext methods"

File: content/en/docs/configuring-sdk/logging.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+This allows your logging implementations to return a new `Logger` that can write additional logging metadata based
+on values present in the context.
+
+## ClientLogMode
+
+By default, service clients will not produce log messages. To configure clients to send log messages for debugging
```
Comment (Doug-AWS) [inline content/en/docs/configuring-sdk/logging.md:38]: "will not" -> "do not"

File: content/en/docs/getting-started.md [hunk 10]
```diff
@@ -33,7 +34,7 @@ command.
 
 The following commands show how to retrieve the standard set of SDK modules that you will use in your application.
```
Comment (Doug-AWS) [inline content/en/docs/getting-started.md:35]: "that you will" -> "to"

**Review by Doug-AWS — CHANGES_REQUESTED** [review]

File: content/en/docs/making-requests.md [hunk 11]
```diff
@@ -1,196 +1,246 @@
...
+})
+```
 
-    // Call ListTables just to see HTTP request/response
-    // The request should have the CustomHeader set to 10
-    _, _ = svc.ListTables(&dynamodb.ListTablesInput{})
+#### Overriding Configuration
+
+`New` can take one or more functional arguments that can mutate a client's configuration `Options` struct.
+This allows you to make specific overrides such a changing the Region or modifying service specific options
```
Comment (Jerry-AWS) [inline content/en/docs/making-requests.md:100]: typo: **such a** -> **such as**

File: content/en/docs/making-requests.md [hunk 12]
```diff
@@ -1,196 +1,246 @@
...
 
-You can call a service operation directly or with its request form. When
-you call a service operation, the SDK synchronously validates the input,
-builds the request, signs it with your credentials, sends it to AWS, and
-then gets a response or an error. In most cases, you can call service
-operations directly.
+### Passing Parameters to a Service Operation
 
-### Calling Operations
+Each service operation method takes a [context.Context](https://golang.org/pkg/context/#Context) values that can be
```
Comment (Jerry-AWS) [inline content/en/docs/making-requests.md:143]: typo: **values** -> **value**

File: content/en/docs/making-requests.md [hunk 13]
```diff
@@ -1,196 +1,246 @@
...
+used for setting request deadlines that will be honored by the SDK. In addition, each service operation will take a
+`<OperationName>Input` struct is found in the service's respective Go package. You pass in API input parameters using
```
Comment (Jerry-AWS) [inline content/en/docs/making-requests.md:145]: typo: **is found** -> **found**

File: content/en/docs/making-requests.md [hunk 14]
```diff
@@ -1,196 +1,246 @@
...
-*OperationName*`Input` and *OperationName*`Output`.
+### Handling Operation Responses {#HandlingOperationResponses}
 
-For more information about the parameters of each method, see the
-service client documentation in the.
+Each service operation has an associated output struct that contains the service's operation response members.
+The output struct follows the following naming pattern `<OperationName>Output`. Some operations may have no members
+defined for their operation output. After calling a service operation, the return `error` argument type should always
+be checked to determine if an error occurred while invoking the service operations. Errors returned can range from
+client-side input validation errors to service-side error responses returned to the client. The operations output
```
Comment (Jerry-AWS) [inline content/en/docs/making-requests.md:218]: typo: **operations** -> **operation's**

File: content/en/docs/making-requests.md [hunk 15]
```diff
@@ -1,196 +1,246 @@
...
+For more information on error handling, including how to inspect for specific error types, see the
+[Handling Errors]({{% ref "handling-errors.md" %}}) documentation.
+
+#### Responses with io.ReadCloser
 
-Some API operations return a response struct that contain a `Body`
-field that is an `io.ReadCloser`. If you're making requests with
-these operations, always be sure to call :code:`Close` on the field.
+Some API operations return a response struct that contain an output member that is an `io.ReadCloser`. If you're making
+requests with these operations, always be sure to call `io.ReadCloser` members `Close` method after you've completed
```
Comment (Jerry-AWS) [inline content/en/docs/making-requests.md:236]: typo: **members** -> **member's**

File: content/en/docs/making-requests.md [hunk 16]
```diff
@@ -207,133 +257,138 @@ if err := decoder.Decode(&myStruct); err != nil {
...
-
-The following example uses the `ListObjectPages` pagination method to
-list up to three pages of object keys from the `ListObject`
-operation. Each page consists of up to 10 keys, which is defined by
-the `MaxKeys` field.
+the output for a token or marker to confirm whether an AWS returned all results from your request.
+If present, you use the token or marker to request the next page of results. Instead of managing these tokens or
+markers, you can use the service package's available paginator types.
+
+Paginator helpers are available for support service operations. Constructing an operation paginator is using operations
```
Comment (Jerry-AWS) [inline content/en/docs/making-requests.md:331]: I think the sentence "Paginator helpers are available for support service operations." can go entirely and the one after that needs work; I am having trouble understanding what it means.

**Review by Jerry-AWS — APPROVED** [review]

**Review by Jerry-AWS — CHANGES_REQUESTED** [review]
Mostly typos and other minor errors.

File: content/en/docs/making-requests.md (hunk 16 above)
Comment (skmcgrail) [reply content/en/docs/making-requests.md:331]: Good catch, clearly I must have walked away and came back to write gibberish :)

File: content/en/docs/making-requests.md [hunk 17]
```diff
@@ -5,193 +5,242 @@ date: "2020-11-12"
...
 
-When you create a service client, you can pass in custom configurations
-so that you don't need to create a session for each configuration. The
-SDK merges the two configurations, overriding session values with your
-custom configuration. For example, in the following snippet, the |S3|
-client uses the `mySession` session but overrides the `Region` field
-with a custom value (``us-west-2``):
+#### Overriding Configuration
+`NewFromConfig` can take one or more functional arguments that can mutate a client's configuration `Options` struct.
+This allows you to make specific overrides such a changing the Region or modifying service specific options such as
```
Comment (Jerry-AWS) [inline content/en/docs/making-requests.md:49]: typo: **such a changing** -> **such as changing**

probably a comma after **Region** would help too

**Review by Jerry-AWS — APPROVED** [review]

**Review by smckenzie23 — APPROVED** [review]
Approving, but want to note a typo we should fix.

"This includes configuring a credential provider. configuring the AWS Region,"

should switch period for comma.

File: content/en/docs/making-requests.md [hunk 18]
```diff
@@ -205,136 +254,173 @@ if err := decoder.Decode(&myStruct); err != nil {
...
+results := make(chan result, 2)
+
+var wg sync.WaitGroup
+wg.Add(2)
+
 go func() {
-    wg.Wait()
-    close(keysCh)
+defer wg.Done()
+    output, err := client.PutObject(context.Background(), &s3.PutObjectInput{
```
Comment (jasdel) [inline content/en/docs/making-requests.md:325]: As a general what do you think abvout using `context.TODO()` here to signal replace user user's context.

File: content/en/docs/making-requests.md [hunk 19]
```diff
@@ -205,136 +254,173 @@ if err := decoder.Decode(&myStruct); err != nil {
...
+        Bucket: aws.String("my-bucket"),
+        Key:    aws.String("foo"),
+        Body: bytes.NewReader([]byte("foo body content")),
```
Comment (jasdel) [inline content/en/docs/making-requests.md:328]: nit could just be `strings.NewReader`

File: content/en/docs/making-requests.md [hunk 20]
```diff
@@ -205,136 +254,173 @@ if err := decoder.Decode(&myStruct); err != nil {
...
-    MaxKeys: aws.Int64(10),
+
+client := s3.NewFromConfig(cfg)
+
+params := &s3.ListObjectsV2Input{
+	Bucket: aws.String("my-bucket"),
 }
+
+pagiantor := s3.NewListObjectsV2Paginator(client, params, func(o *ListObjectsV2PaginatorOptions) {
+	o.Limit = aws.Int64(10)
```
Comment (jasdel) [inline content/en/docs/making-requests.md:401]: is this should be a value not boxed. S3's paginator limit is unboxed.

**Review by jasdel — CHANGES_REQUESTED** [review]

File: content/en/docs/making-requests.md (hunk 19 above)
Comment (skotambkar) [reply content/en/docs/making-requests.md:328]: nit: add import "bytes" / "strings"

**Pull request merged**